docs/site/tutorials/authentication/Authentication-Tutorial.md

```diff
@@ -0,0 +1,974 @@
+application using the
+[JSON Web Token (JWT)](https://en.wikipedia.org/wiki/JSON_Web_Token) approach.
+
+### Installing @loopback/authentication
+
+The `loopback4-example-shopping` application **already** has the
+`@loopback/authentication` dependency set up in its **package.json**
+
+It was installed as a project dependency by performing:
+
+```sh
+npm install --save @loopback/authentication
+```
+
+### Adding the AuthenticationComponent to the Application
+
+The core of authentication framework is found in the
+[AuthenticationComponent](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/authentication.component.ts),
+so it is important to add the component in the `ShoppingApplication` class in
+[loopback4-example-shopping/packages/shopping/src/application.ts](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/application.ts).
+
+```ts
+import {
+  AuthenticationComponent
+} from '@loopback/authentication';
+
+export class ShoppingApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options?: ApplicationConfig) {
+    super(options);
+
+    // ...
+
+    // Bind authentication component related elements
+    this.component(AuthenticationComponent);
+
+    // ...
+```
+
+### Securing an Endpoint with the Authentication Decorator
+
+Securing your application's API endpoints is done by decorating controller
+functions with the
+[Authentication Decorator](../../decorators/Decorators_authenticate.md).
+
+The decorator's syntax is:
+
+```ts
+@authenticate(strategyName: string, options?: object)
+```
+
+In the `loopback4-example-shopping` application, there is only one endpoint that
+is secured.
+
+In the `UserController` class in the
+[loopback4-example-shopping/packages/shopping/src/controllers/user.controller.ts](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts),
+a user can print out his/her user profile by performing a `GET` request on the
+`/users/me` endpoint which is handled by the `printCurrentUser()` function.
+
+```ts
+  // ...
+
+  @get('/users/me', {
+    responses: {
+      '200': {
+        description: 'The current user profile',
+        content: {
+          'application/json': {
+            schema: UserProfileSchema,
+          },
+        },
+      },
+    },
+  })
+  @authenticate('jwt')
+  async printCurrentUser(
+    @inject(AuthenticationBindings.CURRENT_USER)
+    currentUserProfile: UserProfile,
+  ): Promise<UserProfile> {
+    return currentUserProfile;
+  }
+
+  // ...
+```
+
+The `/users/me` endpoint is decorated with
+
+```ts
+@authenticate('jwt')
+```
+
+and authentication will only succeed if a valid JWT token is provided in the
+`Authorization` header of the request.
+
+Basically, the
+[AuthenticateFn](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/providers/auth-action.provider.ts)
```

Comment:
@emonddr I think a more accurate word here would be [Authenticate action](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/providers/auth-action.provider.ts)

That action, exported as a provider, returns a function in type `AuthenticateFn`, so `AuthenticateFn` is actually a type. The action name should still be "authenticate"